cpp/include/cudf/table/row_operators.cuh

```diff
@@ -138,6 +139,52 @@ __device__ bool equality_compare(Element const lhs, Element const rhs) {
     return lhs == rhs;
 }
 
+/**---------------------------------------------------------------------------*
+ * @brief Computes whether an element in a column is equal to a scalar value
+ *
+ * @tparam Element identifies the type of the element
+ * @tparam has_nulls Indicates the potential for null values in the column
+ *---------------------------------------------------------------------------**/
+template <typename Element, bool has_nulls = true>
+struct compare_with_value {
+  using ScalarDeviceType = cudf::experimental::scalar_device_type_t<Element>;
+  /**---------------------------------------------------------------------------*
+   * @brief Construct a function object for comparing an element of a column
+   * to a scalar
+   *
+   * @param col The column
+   * @param val The scalar value to compare with
+   * @param nulls_are_equal Indicates if two null elements are treated as
+   *equivalent
+   *---------------------------------------------------------------------------**/
+  compare_with_value(column_device_view col, const ScalarDeviceType val, bool nulls_are_equal)
+    : col{col}, val{val}, nulls_are_equal{nulls_are_equal} {}
+
+  /**---------------------------------------------------------------------------*
+   * @brief Checks whether the element at `index` in the column is equal
+   * to the scalar value
+   *
+   * @param index The index of element in the column
+   * @return `true` if element from the column compares equal to the scalar value
+   *---------------------------------------------------------------------------**/
+  __device__ bool operator()(size_type index) noexcept {
+    if (has_nulls) {
```

Comment:
I would combine the first two conditions into a single `if` with `&&`.

Reply:
Done